### PR TITLE
R77

### PR DESCRIPTION
**Pull request by alexanderdean**

### CODE REVIEW

**Comment by alexanderdean** [comment]
This looks great Fred! Can you add the CHANGELOG in too?

**Comment by fblundun** [comment]
Done!

File: 3-enrich/emr-etl-runner/lib/snowplow-emr-etl-runner/runner.rb [hunk 1]
```diff
@@ -127,6 +127,15 @@ def validate_and_coalesce(args, config)
           raise ConfigError, "collector_format '%s' not supported" % input_collector_format
         end
 
+        if input_collector_format == 'thrift'
+          if args[:skip].include?('s3distcp')
+            raise ConfigError, "Cannot process Thrift events with --skip s3distcp"
+          end
+          if config[:emr][:ami_version].start_with?('2')
```
Comment (alexanderdean) [inline 3-enrich/emr-etl-runner/lib/snowplow-emr-etl-runner/runner.rb]: Should be `[:aws][:emr]...` @fblundun

File: 3-enrich/emr-etl-runner/config/config.yml.sample [hunk 2]
```diff
@@ -34,8 +34,8 @@ aws:
     ec2_key_name: ADD HERE
     bootstrap: []           # Set this to specify custom boostrap actions. Leave empty otherwise
     software:
-      hbase:                # To launch on cluster, provide version, "0.92.0", keep quotes
```
Comment (alexanderdean) [inline 3-enrich/emr-etl-runner/config/config.yml.sample]: Shouldn't ami_version be bumped to 4.3.0 too?

Comment (fblundun) [inline 3-enrich/emr-etl-runner/config/config.yml.sample]: Good point! I will change this now.

**Comment by alexanderdean** [comment]
@jbeemster, can you update the CHANGELOG with all your commits?

@chuwy, can you update the CHANGELOG with your commits when done?

**Comment by chuwy** [comment]
Done.

**Comment by jbeemster** [comment]
Done

**Comment by alexanderdean** [comment]
Thanks both!

**Comment by alexanderdean** [comment]
I think we're getting very close with this branch - @fblundun can you move this commit 60d0ba065ed0dbb7e8d541b46e516c9fba2d4cc6 into the two preceding commits which have the correct issues?

**Comment by fblundun** [comment]
Done!

**Comment by alexanderdean** [comment]
Thanks!

File: 4-storage/storage-loader/bin/snowplow-runner-and-loader.sh [hunk 3]
```diff
@@ -25,7 +26,7 @@ source ${rvm_path}/scripts/rvm
 
 # Run the ETL job on EMR
 export BUNDLE_GEMFILE=${RUNNER_PATH}/Gemfile
```
Comment (alexanderdean) [inline 4-storage/storage-loader/bin/snowplow-runner-and-loader.sh]: Why are we exporting BUNDLE_GEMFILE for JRuby?

File: 4-storage/storage-loader/bin/snowplow-runner-and-loader.sh [hunk 4]
```diff
@@ -42,3 +43,4 @@ fi
 # If all okay, run the storage load too
 export BUNDLE_GEMFILE=${LOADER_PATH}/Gemfile
 bundle exec ${LOADER_PATH}/bin/snowplow-storage-loader --config ${LOADER_CONFIG}
+./${LOADER_PATH} --config ${LOADER_CONFIG}
```
Comment (alexanderdean) [inline 4-storage/storage-loader/bin/snowplow-runner-and-loader.sh]: Why are we running the Ruby and then the JRuby versions?

**Comment by alexanderdean** [comment]
@fblundun - any ideas why Travis is not passing?

File: 4-storage/storage-loader/bin/snowplow-runner-and-loader.sh (hunk 4 above)
Comment (fblundun) [inline 4-storage/storage-loader/bin/snowplow-runner-and-loader.sh]: Good spot! I'll remove the `bundle exec` line.

File: 4-storage/storage-loader/bin/snowplow-runner-and-loader.sh (hunk 3 above)
Comment (fblundun) [inline 4-storage/storage-loader/bin/snowplow-runner-and-loader.sh]: I'll remove it.

File: 4-storage/storage-loader/bin/snowplow-runner-and-loader.sh (hunk 4 above)
Comment (alexanderdean) [inline 4-storage/storage-loader/bin/snowplow-runner-and-loader.sh]: We should be able to remote the `export` line too...

**Comment by fblundun** [comment]
There were some tickets created after the CHANGELOG was written. So we will need to do one more rebase just before merging.

**Comment by alexanderdean** [comment]
Yep agreed

File: 4-storage/storage-loader/lib/snowplow-storage-loader/redshift_loader.rb [hunk 5]
```diff
@@ -103,14 +96,19 @@ def load_events_and_shredded_types(config, target, snowplow_tracking_enabled)
...
+            raise DatabaseLoadError, Sanitization.sanitize_message("#{vacuum_status[1]} error executing VACUUM statements: #{vacuum_status[0]}: #{vacuum_status[2]}", credentials)
+          end
+        end
+
+        # ANALYZE statements should be executed after VACUUM statements.
+        unless config[:skip].include?('analyze')
+          analyze_statements = [build_analyze_statement(target[:table])] + shredded_statements.map(&:analyze).uniq
+          analyze_status = PostgresLoader.execute_transaction(target, analyze_statements)
+          unless analyze_status == []
+            raise DatabaseLoadError, Sanitization.sanitize_message("#{analyze_status[1]} error executing ANALYZE statements: #{analyze_status[0]}: #{analyze_status[2]}")
```
Comment (alexanderdean) [inline 4-storage/storage-loader/lib/snowplow-storage-loader/redshift_loader.rb]: @fblundun this call to `sanitize_message` is missing the second argument, to `credentials`

**Comment by alexanderdean** [comment]
@fblundun I fixed the missing `credentials` in the arg call

**Pull request merged**